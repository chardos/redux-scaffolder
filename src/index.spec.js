const rsk = require('./index.js');
const fs = require('fs');

jest.spyOn(fs, 'writeFile').mockReturnValue(Promise.resolve(null));

describe('When running rsk in the command line', () => {
  beforeEach(() => {
    process.argv.splice(2);
  })

  it('stateless functional component should match the snapshot', async () => {
    process.argv.push('sfc', 'potato');
    const output = await rsk();
    expect(output).toMatchSnapshot()
  })

  it('class component should match the snapshot', async () => {
    process.argv.push('cc', 'potato');
    console.log('process.argv', process.argv);
    const output = await rsk();
    expect(output).toMatchSnapshot()
  })

})