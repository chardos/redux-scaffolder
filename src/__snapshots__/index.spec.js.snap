// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When running rsk in the command line class component should match the snapshot 1`] = `
"import React from \\"react\\";

class Potato extends React.Component {
  render() {
    return <div>Potato</div>;
  }
}

export default Potato;
"
`;

exports[`When running rsk in the command line setup store should match the snapshot 1`] = `
"
  import { createStore, compose, combineReducers, applyMiddleware } from \\"redux\\";
  import reducers from './ducks';

  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const rootReducer = combineReducers(reducers);

  const store = createStore(rootReducer, composeEnhancers(
      applyMiddleware(thunk)
  ));

  export default store;
"
`;

exports[`When running rsk in the command line stateless functional component should match the snapshot 1`] = `
"import React from \\"react\\";

const Potato = () => {
  return <div>Potato</div>;
};

export default Potato;
"
`;
